app.rb

```diff
@@ -0,0 +1,112 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  attr_reader :books, :people, :rentals, :classroom
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+    @classroom = Classroom.new('college')
+  end
+
+  # Method to list all books
+  def list_all_books()
+    puts 'All Books:'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  # Method to list all people
+  def list_all_people()
+    puts 'All People:'
+    @people.each do |person|
+      puts "ID: #{person.id}, Name: #{person.name}"
+    end
+  end
+
+  # Method to create a person
+  def create_person()
+    puts 'Enter person name:'
+    name = gets.chomp
+    puts 'Enter person age:'
+    age = gets.chomp.to_i
+    puts 'Is it a student? (y/n):'
+    is_student = gets.chomp.downcase == 'y'
+
+    if is_student
+      puts 'Enter student classroom:'
+      classroom = gets.chomp
+      person = Student.new(age, name, classroom, parent_permission: true)
+    else
+      puts 'Enter teacher specialization:'
+      gets.chomp
+      person = Person.new(age, name: name, parent_permission: true)
+    end
+
+    people << person
+    puts 'Person created successfully.'
+  end
+
+  # Method to create a book
+  def create_book
+    puts 'Enter book title:'
+    title = gets.chomp
+    puts 'Enter book author:'
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully.'
+  end
+
+  # Method to create a rental
+  def create_rental
+    puts 'Enter person ID:'
+    person_id = gets.chomp.to_i
+    person = people.find { |p| p.id == person_id }
+
+    unless person
+      puts 'Person not found.'
+      return
+    end
+
+    puts 'Enter book title:'
+    book_title = gets.chomp
+    book = books.find { |b| b.title == book_title }
+
+    unless book
+      puts 'Book not found.'
+      return
+    end
+
+    puts 'Enter rental date:'
+    date = gets.chomp
+
+    rental = Rental.new(date, book, person)
+    rentals << rental
+    puts 'Rental created successfully.'
+  end
```

Comment:
- [ ] The app is not throwing the errors as before. Well done 🥇 However, I noticed that your current implementation of creating a rental is different from the given video 📸 After selecting option `5`, the app is asking for the `ID of the person` instead of displaying a list of books and people list. Also, please note that we need to put the number (index) not the ID Kindly fix your current implementation according to the given video ✨⛏ as mentioned in the project requirements

Current UI | Expected UI
---| ---|
![image](https://github.com/Naledi-Dikgale/oop-school-library/assets/106119105/6b58f3a5-bc46-4249-ad0b-c85c3730d2fe) | ![image](https://github.com/Naledi-Dikgale/oop-school-library/assets/106119105/466381c9-05ec-4b9f-88a0-1acec16143ec)

- [ ] After adding the person's ID following the current UI, the app is asking to enter the book title, and displays a **_Book not found._** message if it is not present ⚠ If the book is present, it is asking for the date. There's no information of the author. Please note that this is not the expected behaviour. Kindly fix the implementation of creating a rental according to the [referenced video](https://www.youtube.com/watch?v=vkkgrhD6aXQ)

Current UI | Expected UI
---| ---|
![image](https://github.com/Naledi-Dikgale/oop-school-library/assets/106119105/5d021d2e-4c0d-4d2c-a974-f14e2f2a5640) | ![image](https://github.com/Naledi-Dikgale/oop-school-library/assets/106119105/c7147b69-2f45-4687-b0c4-18f9336e1a13)


The expected behaviour is 👇
1. After selecting `5 (Create a rental)` from the list, a list of the present books should appear
2. After selecting the number (index) from the list, a list of the people should appear
3.  After selecting the number (not ID), we can add a date of rental, and rental should be created successfully